pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,693 @@
+            text_options=None (dict): any matplotlib keywords for the text
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=1)
+
+            drawer.box_gate(layer=0, wires=(0, 1), text="CY")
+
+        .. figure:: ../../_static/drawer/box_gates.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        This method can accept two different sets of design keywords. ``box_options`` takes
+        `Rectangle keywords <https://matplotlib.org/stable/api/_as_gen/matplotlib.patches.Rectangle.html>`_
+        , and ``text_options`` accepts
+        `Matplotlib Text keywords <https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.text.html>`_ .
+
+        .. code-block:: python
+
+            box_options = {'facecolor': 'lightcoral', 'edgecolor': 'maroon', 'linewidth': 5}
+            text_options = {'fontsize': 'xx-large', 'color':'maroon'}
+
+            drawer = MPLDrawer(n_wires=2, n_layers=1)
+
+            drawer.box_gate(layer=0, wires=(0, 1), text="CY",
+                box_options=box_options, text_options=text_options)
+
+        .. figure:: ../../_static/drawer/box_gates_formatted.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        if box_options is None:
+            box_options = {}
+        if "zorder" not in box_options:
+            box_options["zorder"] = 2
+
+        new_text_options = {"zorder": 3, "ha": "center", "va": "center", "fontsize": "x-large"}
+        if text_options is not None:
+            new_text_options.update(text_options)
+
+        wires = _to_tuple(wires)
+
+        box_min = min(wires)
+        box_max = max(wires)
+        box_len = box_max - box_min
+        box_center = (box_max + box_min) / 2.0
+
+        box = plt.Rectangle(
+            (layer - self._box_dx - extra_width / 2, box_min - self._box_dx),
+            2 * self._box_dx + extra_width,
+            (box_len + 2 * self._box_dx),
+            **box_options,
+        )
+        self._ax.add_patch(box)
+
+        self._ax.text(
+            layer,
+            box_center,
+            text,
+            **new_text_options,
+        )
+
+    def ctrl(self, layer, wires, wires_target=None, control_values=None, options=None):
+        """Add an arbitrary number of control wires
+
+        Args:
+            layer (int): the layer to draw the object in
+            wires (Union[int, Iterable[int]]): set of wires to control on
+
+        Keyword Args:
+            wires_target=None (Union[int, Iterable[int]]): target wires. Used to determine min
+                and max wires for the vertical line
+            control_values=None (Union[bool, Iterable[bool]]): for each control wire, denotes whether to control
+                on ``False=0`` or ``True=1``
+            options=None (dict): dictionary of options.  The only supported keys are ``'color'``, ``'linewidth'``,
```

Comment:
I'd update this line slightly to describe better that these are (Matplotlib) customization options, specifically. Saying that they are a "dictionary of options" is a bit superfluous since the typing already conveys it. Something like:
```suggestion
            options=None (dict): Matplotlib keywords. The only supported keys are ``'color'``, ``'linewidth'``,
```